docs/components/FSXABaseComponent.md

```diff
@@ -0,0 +1,128 @@
+[Readme](../../README.md) » [BaseComponents](../README.md) » FSXABaseComponent
+
+# FSXABaseComponent
+
+1. [Introduction](#introduction)
+2. [Usage](#usage)
+3. [Properties](#properties)
+4. [Methods](#methods)
+
+## Introduction
+
+The FSXABaseComponent is there to have a basic component to be able to implement further components.
+Thereby different attributes and methods are given, which should simplify the development significantly. These are described in this section.
+
+## Usage
+
+To use this base component a new class has to be created which extends the `FSXABaseComponent`.
+
+```tsx
+@Component
+class Component extends FSXABaseComponent {}
+```
+
+<hr>
+If you want to have type support for your props you can provide an interface and pass it as an generic like here the `Props` interface.
+
+```tsx
+@Component
+class Component extends FSXABaseComponent<Props> {}
+```
+
+<hr>
+If you want additionally type support for your events, you can pass a second interface.
+
+```tsx
+@Component
+class LayoutComponent extends FSXABaseComponent<Props, EventsWithOn> {}
+```
+
+Note: If you only want type support for your events and not the props you have to pass an empty interface `{}` as the data generic.
+
```

Comment:
to be more precise: empty or _no_ interface, right?

Reply:
Nope, is has to be empty. You can see this like parameter in a function, you have to pass the first parameter to be able to pass the second, even if the first one is empty.